src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp

```diff
@@ -118,6 +118,8 @@ bool TypeLookupManager::init(
         protocols->mp_participantImpl->getRTPSParticipantAttributes().allocation.locators.max_unicast_locators,
         protocols->mp_participantImpl->getRTPSParticipantAttributes().allocation.locators.max_multicast_locators);
 
+    auto properties = participant_->getAttributes().properties;
+    type_propagation_ = utils::to_type_propagation(properties);
```

Comment:
This pattern repeats a lot. We could add a method in `RTPSParticipantImpl` to get the type propagation.